src/type-checker.cc

```diff
@@ -255,23 +255,23 @@ Result TypeChecker::PopAndCheck3Types(Type expected1,
   return result;
 }
 
-Result TypeChecker::CheckOpcode1(Opcode opcode) {
-  Result result = PopAndCheck1Type(opcode.GetParamType1(), opcode.GetName());
+Result TypeChecker::CheckOpcode1(Opcode opcode, const Limits *limits) {
```

Comment:
Maybe add an overload, or use a default value of `limits=nullptr`

Reply:
I tend to use default values sparingly, since I wanted to make sure I considered all the cases where it needs an arg. But I guess I can add that after some further tests :)